packages/api-clients/open-api/m2mGatewayApi.yml

```
@@ -17207,6 +17207,117 @@ paths:
               $ref: "#/components/headers/RateLimitRemainingHeader"
             X-Rate-Limit-Interval:
               $ref: "#/components/headers/RateLimitIntervalHeader"
+  /purposeTemplates/{purposeTemplateId}/linkEservices:
+    parameters:
+      - name: purposeTemplateId
+        in: path
+        description: The Purpose Template ID
+        required: true
+        schema:
+          type: string
+          format: uuid
+    post:
+      tags:
+        - purposeTemplates
+      summary: Link E-Services to Purpose Template
+      description: Link E-Services to a Purpose Template that is in Draft or Active state
+      operationId: linkEServicesToPurposeTemplate
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                eserviceIds:
+                  type: array
+                  items:
+                    type: string
+                    format: uuid
+                  minItems: 1
+              required:
+                - eserviceIds
+        required: true
+      responses:
+        "200":
+          description: E-Services linked to Purpose Template
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/EServiceDescriptorPurposeTemplate"
```

Endpoints that link two resources return 204 no content in this spec (see other linking/unlinking methods)

```suggestion
        "204":
          description: E-Service linked to Purpose Template
```